### ISSUE TITLE
Error following get form

### ISSUE THREAD

**Issue by waiting-for-dev** [boydm/phoenix_integration#11]
Hi!

I'm trying to follow a search form. It is a form with `:get` method pointing to index action.

If I do:

```elixir
conn
|> get(event_path(conn, :index))
|> follow_form(%{search: %{     
  source_eq: "web"     
}}, identifier: event_path(conn, :index))
```

It gets confused with the `create` action:

```
** (Phoenix.ActionClauseError) could not find a matching Eventbox.Web.EventController.create clause       
     to process request. This typically happens when there is a                                                
     parameter mismatch but may also happen when any of the other
     action arguments do not match. The request parameters are:
                               
       %{"_utf8" => "✓", "search" => %{"category_id_eq" => nil, "client_type_id_eq" => nil, "lounge_id_eq" => nil, "people_eq" => nil, "source_eq" =>
 "web", "start_datetime_gte" => nil, "start_datetime_lte" => nil, "status_eq" => nil, "subcategory_id_eq" => nil}}
```

While if I specify the method:

```elixir
conn
|> sign_in_user(user)
|> get(event_path(conn, :index))
|> follow_form(%{search: %{
  source_eq: "web"
}}, identifier: event_path(conn, :index), method: :get)
```

It doesn't find the form:

```
** (RuntimeError) Failed to find form "/events", :get in the response
Expected to find a form with action="/events"
```

Thanks!

**Comment by ZombieHarvester** [comment]
Same here. Docs say that `follow_form requests the form’s action`, but it doesn't mention that it shouldn't be `get`

this is how my search form looks like (in haml)
`- form_tag search_path(@conn, :index, @current_account), method: :get do`

**Comment by boydm** [comment]
I'm super busy right now getting ready for talk at ElixirConf. Will look at it some time after I free up. I vaguely remember something like this from the past, so I suspect there is already some way to do it (may not be tho...). If you figure it out before I get to it (will be at least a week), let me know!

**Comment by boydm** [comment]
Can you take a look and let me know if this is fixed for you in the current master. Thanks.

**Comment by waiting-for-dev** [comment]
Not yet :(

```elixir
** (RuntimeError) Failed to find form "/clients", :get in the response                                                                           
     Expected to find a form with action="/clients"                                                                                                   
     code: |> follow_form(%{search: %{                                                                                                                
     stacktrace:                                                                                                                                      
       (phoenix_integration) lib/phoenix_integration/requests.ex:572: PhoenixIntegration.Requests.find_html_form/4                                    
       (phoenix_integration) lib/phoenix_integration/requests.ex:287: PhoenixIntegration.Requests.submit_form/3                                       
       (phoenix_integration) lib/phoenix_integration/requests.ex:340: PhoenixIntegration.Requests.follow_form/3
```

**Comment by boydm** [comment]
Getting around to this properly. Sorry it took so long.

Trying to write a test that repos, but it is working. Can you attach include a snippet of the html of the form?

**Comment by boydm** [comment]
Never mind. I see it on the site I'm building. Search form is the right example...

**Comment by boydm** [comment]
This is fixed in master. Wasn't getting the form's method correctly in the case of a get.

**Comment by boydm** [comment]
The fix is published to hex in version 0.4

**Comment by waiting-for-dev** [comment]
Thanks @boydm :+1:
